services/user/Sites/src/Sites.cpp

```diff
@@ -53,6 +80,188 @@ namespace SystemService
             target.resize(pos);
          return target;
       }
+
+      /*
+        script-src:
+          * `unsafe-eval` is needed for WebAssembly.compile()
+          * `unsafe-inline` is needed for inline <script> tags
+          * `blob:` for loading dynamically generated modules
+          * `https:` for dynamically loading libs from CDNs
+        font-src:
+          * `https:` for dynamically loading fonts from CDNs
+        frame-src:
+          * `*` is needed to allow embedding supervisor
+        connect-src:
+          * `blob:` for fetch-compiling blob urls
+          * `*` allows fetching plugins, as well as websocket connections
+      */
+      const std::string DEFAULT_CSP_HEADER =                               //
+          "default-src 'self';"                                            //
+          "font-src 'self' https:;"                                        //
+          "script-src 'self' 'unsafe-eval' 'unsafe-inline' blob: https:;"  //
+          "img-src *;"                                                     //
+          "style-src 'self' 'unsafe-inline';"                              //
+          "frame-src *;"                                                   //
+          "connect-src * blob:;"                                           //
+          ;
+
+      // Accepted content encodings
+      constexpr std::string_view validEncodings[] = {
+          // KEEP SORTED
+          "br",
+          "gzip",
+      };
+
+      // Content-coding-identifier, e.g. "br", "gzip", etc.
+      using cci = std::string;
+      // The account that holds the service code implementing the DecompressorInterface
+      using decompressor_account                                              = std::string;
+      const std::array<std::pair<cci, decompressor_account>, 1> decompressors = {
+          {{"br", "psi-brotli"}}  //
+      };
+
+      // Remove leading and trailing whitespace from a string
+      void trimSpaceAround(std::string& str)
```

Comment:
We do have boost available: [boost::algorithm::trim](https://www.boost.org/doc/libs/1_86_0/doc/html/doxygen/trim_8hpp_1a43176b48fed44bc161b5ed635bea9cbe.html).

Reply:
Ooh, thank you